src/libNode/Node.cpp

```diff
@@ -2408,6 +2427,199 @@ bool Node::ProcessDoRejoin(const bytes& message, unsigned int offset,
   return true;
 }
 
+// This feature is only available to shard guard node. This allows guard node to
+// change it's network information (IP and/or port).
+// Pre-condition: Must still have access to existing public and private keypair
+bool Node::UpdateShardGuardIdentity() {
+  LOG_MARKER();
+
+  if (!GUARD_MODE) {
+    LOG_GENERAL(
+        WARNING,
+        "Not in guard mode. Unable to update shard guard network info.");
+    return false;
+  }
+
+  if (!Guard::GetInstance().IsNodeInShardGuardList(
+          m_mediator.m_selfKey.second)) {
+    LOG_GENERAL(WARNING,
```

Comment:
To avoid non-guard miner from unnecessarily receiving this message (which might cause confusion), I suggest to just return false here and instead add a "Current node is a shard node. Updating network info." log in line 2450.